### PR TITLE
[Performance]: Ruff

### PR DESCRIPTION
**Pull request by VIKTORVAV99**
## Description

This PR will replace black, flake8 and isort with ruff for a massive performance increase.

> [!Note]
> PR turned out to be bigger than expected, let me know if I should split things up 🙂

### CODE REVIEW

**Comment by madsnedergaard** [comment]
Yaaay, looking forward to this 😍

**Comment by VIKTORVAV99** [comment]
> Yaaay, looking forward to this 😍

It's turning out to be a bit bigger PR than I initially expected so let me know if you want me to split it up a bit 🙂

File: scripts/tooling.py [hunk 1]
```diff
@@ -20,26 +20,27 @@ def _run(cmd: str):
...
+    _run("ruff format .")
 
 
-def lint():
-    _run(
-        "flake8 electricitymap tests parsers --count --select=F401,E901,E999,F821,F822,F823 --show-source --statistics"
-    )
-    _run("black --check .")
-    _run("isort -c .")
+def pylint():
```
Comment (VIKTORVAV99) [inline scripts/tooling.py:27]: This is crazy slow, do we actually need it or can we remove it?

I looked at the pylint rules added to Ruff so far but they don't seem to be available. Check here: https://docs.astral.sh/ruff/rules/#pylint-pl

Comment (madsnedergaard) [reply scripts/tooling.py:27]: cc @FelixDQ and @unitrium, think we can remove it now that we use ruff?

**Review by madsnedergaard — COMMENTED** [review]
Looks good to me, leaving final review to @unitrium :)

File: scripts/tooling.py (hunk 1 above)
Comment (FelixDQ) [reply scripts/tooling.py:27]: Yes, we have removed pylint on the internal repo in favor of ruff. The rules don't overlap 100%, but we believe that the rules from ruff are sufficient. So feel free to remove pylint from contrib as well.

Comment (VIKTORVAV99) [reply scripts/tooling.py:27]: Thanks! Will do so right away!

**Review by madsnedergaard — APPROVED** [review]
Let's just get this merged, it would be so nice to get rid of isort and friends!

**Pull request merged**